fix fsdp device mesh depenency issue

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #121061

as reported in https://github.com/pytorch/torchtrain/pull/103

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang

wanchaol committed:
fix fsdp device mesh depenency issue

as reported in https://github.com/pytorch/torchtrain/pull/103

[ghstack-poisoned]

awgu commented on torch/distributed/fsdp/_common_utils.py:
```diff
@@ -28,7 +28,7 @@
 import torch.nn as nn
 from torch.distributed._composable_state import _get_module_state, _State
-from torch.distributed._tensor.device_mesh import DeviceMesh
+from torch.distributed.device_mesh import DeviceMesh
```
The old import should still be valid 🤔 : https://github.com/pytorch/pytorch/blob/f7a2bae0ac4f4c3b594619c1e39867695501735d/torch/distributed/_tensor/__init__.py#L12

wanchaol replied:
Yeah it's quite weird, this poping up some recursively imports issue in some CI and that's the motivation for the fix https://github.com/pytorch/torchtrain/actions/runs/8118646813/job/22193278282?pr=103

awgu reviewed (approved):
Stamp to unblock

awgu commented:
Check lint before landing if choosing to land

wanchaol committed:
Update on "fix fsdp device mesh depenency issue"


as reported in https://github.com/pytorch/torchtrain/pull/103

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

wanchaol commented:
@pytorchbot merge

wanchaol commented:
@pytorchbot merge -i
